### PR TITLE
chore: release 0.38.0

### PR DESCRIPTION
**Pull request by thbcmlowk**
### Changed

- BREAKING: Rename `model_loading.download_shared_state` to `model_loading.download_train_shared_state` ([#143](https://github.com/Substra/substrafl/pull/143))
- BREAKING: Rename `model_loading.download_aggregated_state` to `model_loading.download_aggregate_shared_state` ([#143](https://github.com/Substra/substrafl/pull/143))
- Numpy < 1.24 in dependencies to keep pickle compatibility with substra-tools numpy version ([#144](https://github.com/Substra/substrafl/pull/144))

### CODE REVIEW

File: substrafl/__version__.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = "0.37.0"
```
Comment (ThibaultFy) [inline substrafl/__version__.py:1]: Il faut mettre rc1 ici avant de release

File: substrafl/__version__.py [hunk 2]
```diff
@@ -1 +1 @@
...
+__version__ = "0.38.0"
```
Comment (ThibaultFy) [inline substrafl/__version__.py:1]:
```suggestion
__version__ = "0.38.0rc1"
```

File: substrafl/__version__.py (hunk 1 above)
Comment (thbcmlowk) [reply substrafl/__version__.py:1]: https://github.com/Substra/substrafl/blob/release-0.38.0-rc1/substrafl/__version__.py

Comment (ThibaultFy) [reply substrafl/__version__.py:1]: Ok bien vu ahah :)

Comment (ThibaultFy) [reply substrafl/__version__.py:1]: Tu peux checker que le package a bien été publié ici https://pypi.org/project/substrafl/#history

File: substrafl/__version__.py (hunk 2 above)
Comment (thbcmlowk) [reply substrafl/__version__.py:1]: Pre-release will be based on [release-0.38.0-rc1](https://github.com/Substra/substrafl/tree/release-0.38.0-rc1). Is that bad?

Comment (ThibaultFy) [reply substrafl/__version__.py:1]: Nop, It's just that usually we iterate on the same branch, but this is as valid as the other method :)

File: substrafl/__version__.py (hunk 1 above)
Comment (thbcmlowk) [reply substrafl/__version__.py:1]: J'allais faire un brave pip install, mais oui 😄

**Review by ThibaultFy — APPROVED** [review]
thx :)

**Pull request merged**